go-controller/pkg/ovn/controller/services/load_balancer.go

```diff
@@ -316,6 +318,19 @@ func buildPerNodeLBs(service *v1.Service, configs []lbConfig, nodes []nodeInfo)
 						targets = switchV6Targets
 					}
 
+					if config.externalTrafficLocal && !config.isLBOrEIP {
```

Comment:
a service can be EIP and nodeport. I think you need to check if the service has nodeport here instead of "not ingress or EIP". wdyt?

Reply:
also in the commit message it says "We are also replacing the vip on NP,EIP&LB per
node switch LB to the special masueradeIP to
support ETP=local on LGW"

which makes me think we want to add VIP for all types listed, but your logic doesn't seem to match that?

Reply:
right the commit message is wrong, I am only replacing this on nodePort svc. Let me change this to reflect the suggestion you made.